src/rocks_record_store.cpp

```diff
@@ -1065,14 +1069,13 @@ namespace mongo {
           _readUntilForOplog(RocksRecoveryUnit::getRocksRecoveryUnit(txn)->getOplogReadTill()) {
         _currentSequenceNumber =
           RocksRecoveryUnit::getRocksRecoveryUnit(txn)->snapshot()->GetSequenceNumber();
-          
-        if (forward && !startIterator.isNull()) {
+
+        if (forward && !startIterator.isNull() && !_readUntilForOplog.isNull()) {
```

Comment:
I added this `&& !_readUntilForOplog.isNull()` part in a try to fix some issues. Now i think it is time to remove this part because it is virtually equal to `forward` condition.